app/controllers/questions_controller.rb

```diff
@@ -1,49 +1,37 @@
 class QuestionsController < ApplicationController
   before_action :set_exam
+  before_action :set_question
 
   def new
     @question = Question.new
-    @question.number = @exam.questions.count + 1
+    set_question_number(@question)
   end
 
   def create
     @question = Question.new(question_params)
-    @question.number = @exam.questions.count + 1
 
+    set_question_number(@question)
     @question.exam = @exam
 
     if @question.save
       flash[:success] = 'Question added!'
       redirect_to edit_exam_question_path(@exam, @question)
     else
-      flash.now[:error] = "Couldn't create question :/"
+      flash.now[:error] = 'Couldn\'t create question :/'
       render 'new'
     end
   end
 
   def show
-    q = Question.find_by(id: params[:id])
-    redirect_to exam_path(@exam) if q.nil?
   end
 
   def edit
-    @question = Question.find_by(id: params[:id])
-    redirect_to exam_path(@exam) if @question.nil?
   end
 
   def update
-    @question = Question.find_by(id: params[:id])
-    redirect_to exam_path(@exam) if @question.nil?
-
     if @question.update(question_params)
-
-      if @question.is_last_question_of_exam?
-        redirect_to edit_exam_question_path(@exam, @question)
-      else
-        previous_number = @question.number + 1
-        next_question = @exam.questions.find_by(number: previous_number)
-        redirect_to edit_exam_question_path(@exam, next_question)
-      end
+      flash[:success] = 'Succesfully updated question!'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.